server/etcdserver/v3_server.go

```diff
@@ -1028,3 +1028,13 @@ func (s *EtcdServer) downgradeCancel(ctx context.Context) (*pb.DowngradeResponse
 	resp := pb.DowngradeResponse{Version: version.Cluster(s.ClusterVersion().String())}
 	return &resp, nil
 }
+
+func (s *EtcdServer) SetDefragActive(active bool) {
```

Comment:
Please don't store more state on *EtcdServer. Please use `healthNotifier` to store state required to calculate whether etcd is healthy or not.

Reply:
done